Backend/src/test/java/ktsnvt/tim1/services/AuthenticationServiceUnitTests.java

```diff
@@ -0,0 +1,144 @@
+@ActiveProfiles("test")
+public class AuthenticationServiceUnitTests {
+
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @MockBean
+    private UserRepository userRepositoryMocked;
+
+    @MockBean
+    private UserMapper userMapperMocked;
+
+    @MockBean
+    private PasswordEncoder passwordEncoderMocked;
+
+    @MockBean
+    private VerificationTokenRepository verificationTokenRepositoryMocked;
+
+
+    @Test
+    @Transactional
+    void register_emailExists_entityAlreadyExistsExceptionThrown(){
+        String email = "mmarkovic@gmail.com";
+        User user = new User(1L, "Marko", "Markovic", "$2y$12$FDOJQfuSrC7UAvBaUaX7UuP9NwZcZGI2joxQcHlzjEMXJBr57XAX6", email, true);
+        UserDTO userToRegister = new UserDTO(null, "Petar", "Petrovic", "KtsNvt1+", email, false);
+        MockHttpServletRequest request = new MockHttpServletRequest();
+
+        Mockito.when(userRepositoryMocked.findByEmail(email)).thenReturn(user);
+
+        assertThrows(EntityAlreadyExistsException.class,() -> authenticationService.register(userToRegister, request));
+
+        verify(userRepositoryMocked, times(1)).findByEmail(email);
+    }
+
+    @Test
+    @Transactional
+    void register_userOk_user() throws EntityAlreadyExistsException, MalformedURLException {
+        String email = "mmarkovic@gmail.com";
+        Long id = 1L;
+        UserDTO userToRegister = new UserDTO(null, "Petar", "Petrovic", "KtsNvt1+", email, false);
+        RegisteredUser registeredUser = new RegisteredUser(null, userToRegister.getFirstName(), userToRegister.getLastName(), "$2y$12$FDOJQfuSrC7UAvBaUaX7UuP9NwZcZGI2joxQcHlzjEMXJBr57XAX6", userToRegister.getEmail(), false);
+        RegisteredUser registeredUserSaved = new RegisteredUser(id, userToRegister.getFirstName(), userToRegister.getLastName(), "$2y$12$FDOJQfuSrC7UAvBaUaX7UuP9NwZcZGI2joxQcHlzjEMXJBr57XAX6", userToRegister.getEmail(), userToRegister.getVerified());
+        UserDTO returnedUserDTO = new UserDTO(id, userToRegister.getFirstName(), userToRegister.getLastName(), null, userToRegister.getEmail(), userToRegister.getVerified());
+        MockHttpServletRequest request = new MockHttpServletRequest();
+
+        Mockito.when(userRepositoryMocked.findByEmail(email)).thenReturn(null);
+        Mockito.when(passwordEncoderMocked.encode(userToRegister.getPassword())).thenReturn("$2y$12$FDOJQfuSrC7UAvBaUaX7UuP9NwZcZGI2joxQcHlzjEMXJBr57XAX6");
+        Mockito.when(userRepositoryMocked.save(registeredUser)).thenReturn(registeredUserSaved);
+        Mockito.when(userMapperMocked.toDTO(registeredUserSaved)).thenReturn(returnedUserDTO);
+
+        UserDTO createdUser = authenticationService.register(userToRegister, request);
+
+        assertEquals(id, createdUser.getId());
+        Assert.assertNull(createdUser.getPassword());
+
+        verify(userMapperMocked, times(1)).toDTO(registeredUserSaved);
+        //verify(userRepositoryMocked, times(1)).save(registeredUser);
```

Comment:
Uncomment this assertion and run test again

Reply:
You can try this: 
`
verify(userRepositoryMocked, times(1)).save(ArgumentMatchers.any(RegisteredUser.class));
`